tools/bug-report/pkg/content/content.go

```diff
@@ -0,0 +1,184 @@
+		"debug/cdsz",
+		"debug/syncz",
+		"debug/config_distribution",
+		"debug/registryz",
+		"debug/endpointz",
+		"debug/endpointShardz",
+		"debug/configz",
+		"debug/resourcesz",
+		"debug/instancesz",
+		"debug/authorizationz",
+		"debug/config_dump",
+		"debug/push_status",
+		"debug/inject",
+	}
+)
+
+// Params contains parameters for running a kubectl fetch command.
+type Params struct {
+	DryRun    bool
+	Verbose   bool
+	Namespace string
+	Pod       string
+	Container string
+}
+
+func retMap(filename, text string, err error) (map[string]string, error) {
+	if err != nil {
+		return nil, err
+	}
+	return map[string]string{
+		filename: text,
+	}, nil
+
+}
+
+// GetK8sResources returns all k8s cluster resources.
+func GetK8sResources(params *Params) (map[string]string, error) {
+	out, err := kubectlcmd.RunCmd("get --all-namespaces "+
+		"all,jobs,ingresses,endpoints,customresourcedefinitions,configmaps,events "+
+		"-o yaml", "", params.DryRun)
+	return retMap("k8s-resources", out, err)
+}
+
+// GetSecrets returns all k8s secrets. If full is set, the secret contents are also returned.
+func GetSecrets(params *Params) (map[string]string, error) {
+	cmdStr := "get secrets --all-namespaces"
+	if params.Verbose {
+		cmdStr += " -o yaml"
+	}
+	out, err := kubectlcmd.RunCmd(cmdStr, "", params.DryRun)
+	return retMap("secrets", out, err)
+}
+
+// GetCRs returns CR contents for all CRDs in the cluster.
+func GetCRs(params *Params) (map[string]string, error) {
+	crds, err := getCRDList(params)
+	if err != nil {
+		return nil, err
+	}
+	out, err := kubectlcmd.RunCmd("get --all-namespaces "+strings.Join(crds, ",")+" -o yaml", "", params.DryRun)
+	return retMap("crs", out, err)
+}
+
+// GetClusterInfo returns the cluster info.
+func GetClusterInfo(params *Params) (map[string]string, error) {
+	out, err := kubectlcmd.RunCmd("cluster-info dump", "", params.DryRun)
+	return retMap("cluster-info", out, err)
+}
+
+// GetDescribePods returns describe pods for istioNamespace.
+func GetDescribePods(params *Params) (map[string]string, error) {
+	if params.Namespace == "" {
+		return nil, fmt.Errorf("getDescribePods requires the Istio namespace")
+	}
+	out, err := kubectlcmd.RunCmd("describe pods", params.Namespace, params.DryRun)
+	return retMap("describe-pods", out, err)
+}
+
+// GetEvents returns events for all namespaces.
+func GetEvents(params *Params) (map[string]string, error) {
+	out, err := kubectlcmd.RunCmd("get events --all-namespaces -o wide", "", params.DryRun)
+	return retMap("events", out, err)
+}
+
+// GetIstiodInfo returns internal Istiod debug info.
+func GetIstiodInfo(params *Params) (map[string]string, error) {
+	if params.Namespace == "" || params.Pod == "" {
+		return nil, fmt.Errorf("getIstiodInfo requires namespace and pod")
+	}
+	ret := make(map[string]string)
+	for _, url := range istiodURLs {
+		out, err := kubectlcmd.Exec(params.Namespace, params.Pod, "discovery", params.DryRun, "curl", `http://localhost:8080/`+url+``)
```

Comment:
if you use pkg/kube you would have an implementation that is actually tested, kept up to date, and far more efficient

Reply:
done